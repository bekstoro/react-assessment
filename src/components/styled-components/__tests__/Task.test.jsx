import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import {
  BlueButton,
  Button,
  Content,
  GreenButton,
  Heading,
  RedButton,
  Task,
} from '@components/styled-components/Task';

describe('Task styled components', () => {
  it('Task renders correctly', () => {
    const { container } = render(<Task />);
    expect(container).toMatchSnapshot();
  });

  it('Heading renders correctly', () => {
    const { container } = render(<Heading />);
    expect(container).toMatchSnapshot();
  });

  it('Button renders correctly', () => {
    const { container } = render(<Button />);
    expect(container).toMatchSnapshot();
  });

  it('RedButton renders correctly', () => {
    const { container } = render(<RedButton />);
    expect(container).toMatchSnapshot();
  });

  it('GreenButton renders correctly', () => {
    const { container } = render(<GreenButton />);
    expect(container).toMatchSnapshot();
  });

  it('BlueButton renders correctly', () => {
    const { container } = render(<BlueButton />);
    expect(container).toMatchSnapshot();
  });

  it('Content renders correctly', () => {
    const { container } = render(<Content />);
    expect(container).toMatchSnapshot();
  });
});
