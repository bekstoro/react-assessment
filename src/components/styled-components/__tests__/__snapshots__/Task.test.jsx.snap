// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task styled components BlueButton renders correctly 1`] = `
.c0 {
  width: 300px;
  height: 70px;
  font-size: 1.1rem;
  text-transform: uppercase;
  border: none;
  border-radius: 5px;
  cursor: pointer;
  color: #fff;
}

.c0:hover {
  opacity: 0.5;
}

.c1 {
  background-color: #5050ff;
}

<div>
  <button
    class="c0 c1"
  />
</div>
`;

exports[`Task styled components Button renders correctly 1`] = `
.c0 {
  width: 300px;
  height: 70px;
  font-size: 1.1rem;
  text-transform: uppercase;
  border: none;
  border-radius: 5px;
  cursor: pointer;
  color: #fff;
}

.c0:hover {
  opacity: 0.5;
}

<div>
  <button
    class="c0"
  />
</div>
`;

exports[`Task styled components Content renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > button:not(:first-child) {
  margin-top: 20px;
}

<div>
  <section
    class="c0"
  />
</div>
`;

exports[`Task styled components GreenButton renders correctly 1`] = `
.c0 {
  width: 300px;
  height: 70px;
  font-size: 1.1rem;
  text-transform: uppercase;
  border: none;
  border-radius: 5px;
  cursor: pointer;
  color: #fff;
}

.c0:hover {
  opacity: 0.5;
}

.c1 {
  background-color: #53ff53;
}

<div>
  <button
    class="c0 c1"
  />
</div>
`;

exports[`Task styled components Heading renders correctly 1`] = `
.c0 {
  color: #216298;
}

<div>
  <h1
    class="c0"
  />
</div>
`;

exports[`Task styled components RedButton renders correctly 1`] = `
.c0 {
  width: 300px;
  height: 70px;
  font-size: 1.1rem;
  text-transform: uppercase;
  border: none;
  border-radius: 5px;
  cursor: pointer;
  color: #fff;
}

.c0:hover {
  opacity: 0.5;
}

.c1 {
  background-color: #ff5252;
}

<div>
  <button
    class="c0 c1"
  />
</div>
`;

exports[`Task styled components Task renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 800px;
  margin: auto;
}

<div>
  <div
    class="c0"
  />
</div>
`;
